### PR TITLE
Instructor: fix bug where session closed email copy is sent to the instructor even when it was not sent to students #8761

### LINKED ISSUE #8761: Instructor: fix bug where session closed email copy is sent to the instructor even when it was not sent to students
Noticed while working on #8447 (Specifically https://github.com/TEAMMATES/teammates/pull/8518#discussion_r179528642)
- **Environment**: `master` branch

**Steps to reproduce:**
- Login as Instructor
- Create a session with questions not answerable to students/ an empty session will do too.

Another easier way to reproduce this bug will be to modify a session in `typicalDataBundle.json` (e.g. `Instructor Feedback Session` and changing the `giverType` and `receiverType` to `INSTRUCTORS` for the questions in that sesssion (`qn1InSession2InCourse2`, `qn2InSession2InCourse2` for above session), and then changing `L190` to
`fsLogic.getFeedbackSession("Instructor feedback session", "idOfTypicalCourse2");` and then running the `EmailGeneratorTest.java`.

https://github.com/TEAMMATES/teammates/blob/8ad5768d4007a1fd5e8eca729975d779840b1126/src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java#L187-L190

The assert will fail for session closed email.
https://github.com/TEAMMATES/teammates/blob/8ad5768d4007a1fd5e8eca729975d779840b1126/src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java#L198-L199

**Expected behaviour**
When the session is closed, no email alerts should be sent to the students, and no email is sent to the students, but in case of session closed emails, a email is sent to all the instructors telling them session closed email has been sent to the students. (Screenshot of the email below)
screenshot 478

As no email is sent to the students, this email should not be sent to the instructor.
**Actual behaviour:**
The email is sent to all the instructors in the course.

**Additional info**

On further looking in the `generateFeedbackSessionClosedEmails` method:
https://github.com/TEAMMATES/teammates/blob/8ad5768d4007a1fd5e8eca729975d779840b1126/src/main/java/teammates/logic/api/EmailGenerator.java#L323-L342

On L338 we are not checking if the email is needed for students or not (Unlike in any other email generator), which is why the students are not receiving the emails (As they shouldn't) but the instructors are still getting the above mentioned email.

just for comparison this is `generateFeedbackSessionPublishedEmails`
https://github.com/TEAMMATES/teammates/blob/8ad5768d4007a1fd5e8eca729975d779840b1126/src/main/java/teammates/logic/api/EmailGenerator.java#L357-L363

### PR DESCRIPTION
**Pull request by sukanta-27**
Fixes #8761 

**Outline of Solution:** Added conditional for instructor similar to other email generators.

### CODE REVIEW

**Comment by sukanta-27** [comment]
This is Ready for review.

File: src/main/java/teammates/logic/api/EmailGenerator.java [hunk 1]
```diff
@@ -335,7 +335,9 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
             log.severe("Course " + session.getCourseId() + " does not exist or "
                     + "session " + session.getFeedbackSessionName() + " does not exist");
         }
-        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(session.getCourseId());
+        List<InstructorAttributes> instructors = isEmailNeededForStudents
+                                                 ? instructorsLogic.getInstructorsForCourse(session.getCourseId())
+                                                 : new ArrayList<InstructorAttributes>();
```
Comment (whipermr5) [inline src/main/java/teammates/logic/api/EmailGenerator.java]: Can use the diamond operator: `new ArrayList<>()`

File: src/main/java/teammates/logic/api/EmailGenerator.java [hunk 2]
```diff
@@ -335,7 +335,9 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
...
         List<StudentAttributes> students = isEmailNeededForStudents
                                            ? studentsLogic.getStudentsForCourse(session.getCourseId())
                                            : new ArrayList<StudentAttributes>();
```
Comment (whipermr5) [inline src/main/java/teammates/logic/api/EmailGenerator.java]: Same here

**Comment by sukanta-27** [comment]
@whipermr5 Should I use diamond operator in other email generating methods too?

**Pull request merged**